Sources/SwiftDocC/Model/SourceLanguage.swift

```diff
@@ -149,4 +149,31 @@ public struct SourceLanguage: Hashable, Codable {
     
     /// The list of programming languages that are known to DocC.
     public static var knownLanguages: [SourceLanguage] = [.swift, .objectiveC, .javaScript, .data, .metal]
+    
+    enum CodingKeys: CodingKey {
+        case name
+        case id
+        case idAliases
+        case linkDisambiguationID
+    }
+    
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: SourceLanguage.CodingKeys.self)
+        
+        let name = try container.decode(String.self, forKey: SourceLanguage.CodingKeys.name)
+        let id = try container.decode(String.self, forKey: SourceLanguage.CodingKeys.id)
+        let idAliases = try container.decodeIfPresent([String].self, forKey: SourceLanguage.CodingKeys.idAliases) ?? []
+        let linkDisambiguationID = try container.decodeIfPresent(String.self, forKey: SourceLanguage.CodingKeys.linkDisambiguationID)
+        
+        self.init(name: name, id: id, idAliases: idAliases, linkDisambiguationID: linkDisambiguationID)
+    }
+    
+    public func encode(to encoder: Encoder) throws {
+        var container = encoder.container(keyedBy: SourceLanguage.CodingKeys.self)
+        
+        try container.encode(self.name, forKey: SourceLanguage.CodingKeys.name)
+        try container.encode(self.id, forKey: SourceLanguage.CodingKeys.id)
+        try container.encodeIfNotEmpty(self.idAliases, forKey: SourceLanguage.CodingKeys.idAliases)
+        try container.encode(self.linkDisambiguationID, forKey: SourceLanguage.CodingKeys.linkDisambiguationID)
```

Comment:
This `encode` implementation creates a situation where some SourceLanguage instances won't round-trip correctly - they may not have a `linkDisambiguationID` to start with, but after round-tripping they'll re-encode with one set. Is this behavior we want?

Reply:
I don't think that's the behavior we want.

Reply:
Hm. What's the downside to this behavior? That _is_ the link disambiguation ID so I don't know that it's incorrect to encoding it as such.

Reply:
The link disambiguation ID is decoded with `decodeIfPresent` but encoded with `encode`. Doesn't that mean that if the original data has no "linkDisambiguationID" key, then the roundtrip data will have an explicit `null` value for the "linkDisambiguationID"? If so, I don't think that's a behavior that we want.

Reply:
The `linkDisambiguationID` is inferred from the `id` so it can never be nil on the `encode`.

Reply:
I suppose we could check if `linkDisambiguationID` is equal to `id` and only encode it if it's different... That would save space and fix the round-tripping behavior.

Reply:
Ah. I missed that. I still prefer when values round-trip correctly but in this case that conflicts with the general encoding of values. 

We could achieve the round-trip behavior by not encoding `linkDisambiguationID` if it's the same as `id` but then that means that it won't be encoded if a `SourceLanguage` was explicitly initialized with the same `linkDisambiguationID` and `id` and that's also a confusing and surprising behavior. 

We could get both behaviors if the `linkDisambiguationID` was privately stored as an optional value—which is what would be encoded—but I'm not sure if it'd be worth it. Ultimately I don't have very strong feelings about the encoding behavior and I'd be fine with either of the 3 solutions.

Reply:
Sounds good. I'd rather land this as-is just to get things up-to-date with the spec and we can follow-up with more refined behavior as needed.